src/webapp/gradebookServer.js

```diff
@@ -280,6 +280,57 @@ app.get('/sections', function(request, response) {
    });
 });
 
+
+//Returns an array of dates for a given section id.  Omits dates when classes are
+// not held.
+app.get('/sectionschedule', function(request, response) {
+   //Decrypt the password recieved from the client.  This is a temporary development
+   //feature, since we don't have ssl set up yet
+   var passwordText = sjcl.decrypt(superSecret, JSON.parse(request.query.password));
+
+   //Connnection parameters for the Postgres client recieved in the request
+   var config = createConnectionParams(request.query.user, request.query.database,
+      passwordText, request.query.host, request.query.port);
+
+   var sectionID = request.query.sectionID;
+
+   var queryText = 'SELECT date FROM significantDate WHERE classesheld = false;';
+   var queryParams = [];
+   var closedDates = [];
+
+   executeQuery(response, config, queryText, queryParams, function(result) {
+      for(row in result.rows) {
+         closedDates.push(
+            {
+               "date": result.rows[row].date
+            }
+         );
+      }
+   });
+
+   queryText = 'SELECT getScheduleDates($1);';
```

Comment:
```suggestion
   queryText = 'SELECT scheduledate::VARCHAR AS date FROM getScheduleDates($1);';
```

The column name used on L317 and L321 has to exactly match the one from the returned table. I believe the best way to guarantee this and keep a clear name is to use a column alias. See comment on the other query in this section for the reason to cast to `VARCHAR`.